storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -960,9 +960,11 @@ bool lock_tables_for_backup(MYSQL *connection, int timeout) {
     return (true);
   }
 
-  msg_ts("Error: LOCK TABLES FOR BACKUP is not supported.\n");
+  tables_locked = true;
```

Comment:
No - the condition to avoid that requires PS. With this, the lock-ddl option is supported when backing up mysql, and the test case I modified passes because of this. But we will still execute FTWRL when we did before, as LIFB only blocks ddl operations, and using only it would break some use cases.

Reply:
OK. It is fine than. One more thing - I don't see UNLOCK INSTANCE.

Reply:
I guess sequence will be
LOCK INSTANCE FOR BACKUP
backup 
UNLOCK TABLES (instead of UNLOCK INSTANCE)

Reply:
Fixed - now we execute unlock instance too.

Reply:
But now UNLOCK TABLES will be unnecessarily invoked when backup was taken with --lock-ddl and without FTWRL. What is the reason for setting `tables_locked` here? If there are places where xtrabackup relies on the knowledge that LTFB or LIFB was invoked, we can check for `tables_locked || instance_locked`...

Reply:
I kept that there to avoid accidentally not unlocking tables we should unlock - as the unlocking part had no conditions depending on `tables_locked`. I've added a condition to it now, with this, we only unlock if tables_locked is set, and looks like that didn't decrese the unlock numbers in the tests.